makeflow/example/example.categories.makeflow

```diff
@@ -0,0 +1,55 @@
+# This is a sample Makeflow script that retrieves an image from the web,
+# creates four variations of it, and then combines them into an animation.
+# It produces the same output as example.makeflow, but it introduces the syntax
+# to group rules in sets, called categories. The computational resources needed
+# for a rule in a category can also be optionally specified.
+
+# A Makeflow script is a subset of the Makefile language (see the manual
+# for more information). For example, for convenience we can define the
+# textual substitutions:
+CURL=/usr/bin/curl
+CONVERT=/usr/bin/convert
+URL=http://ccl.cse.nd.edu/images/capitol.jpg
+
+# We specify the set of inputs and outputs. This is not required, but strongly
+# recommended.
+MAKEFLOW_INPUTS=$CURL $CONVERT
+MAKEFLOW_OUTPUTS=capitol.montage.gif
+
+# Similar rules can be arbitrarily labeled to form sets, called categories. In
+# this example we define three categories: preprocess, analysis, and merge.
+# Rules belong to the category of the most recent CATEGORY statement.
+CATEGORY=merge
+
+# Resources needed for a single task in category may be included (optional).
+# MEMORY and DISK in MB.
+CORES=1
+MEMORY=50
+DISK=100
+capitol.montage.gif: capitol.jpg capitol.90.jpg capitol.180.jpg capitol.270.jpg capitol.360.jpg
+	$CONVERT -delay 10 -loop 0 capitol.jpg capitol.90.jpg capitol.180.jpg capitol.270.jpg capitol.360.jpg capitol.270.jpg capitol.180.jpg capitol.90.jpg capitol.montage.gif
+
+CATEGORY=analysis
+CORES=1
+MEMORY=10
+DISK=20
+capitol.90.jpg: capitol.jpg
+	$CONVERT -swirl 90 capitol.jpg capitol.90.jpg
+
+capitol.180.jpg: capitol.jpg
+	$CONVERT -swirl 180 capitol.jpg capitol.180.jpg
+
+capitol.270.jpg: capitol.jpg
+	$CONVERT -swirl 270 capitol.jpg capitol.270.jpg
+
+capitol.360.jpg: capitol.jpg
+	$CONVERT -swirl 360 capitol.jpg capitol.360.jpg
+
+# If a rule is preceded by LOCAL, it executes at the local site.
+CATEGORY=preprocess
+CORES=1
+MEMORY=10
+DISK=20
+capitol.jpg:
```

Comment:
So is the plan right now to not add the `curl` and `convert` dependencies?

I think it would be great if we had another example that dealt with distributing the locally installed `curl` and `convert` along with its implicit dependencies. I believe we would use `starch` for that right?